pkg/kubectl/cmd/exec.go

```diff
@@ -161,6 +162,10 @@ func (p *ExecOptions) Complete(f *cmdutil.Factory, cmd *cobra.Command, argsIn []
 		}
 	}
 
+	if strings.HasPrefix(p.PodName, "pod/") {
```

Comment:
Not like this.  You want to use a RESTMapper (or at least part of one) to coerce the name to its closest available `GroupResource` and then match based on that..  @fabianofranz can you point him at the code we use in origin.